src/engineio/transports/polling.rs

```diff
@@ -87,34 +87,55 @@ impl Transport for PollingTransport {
     }
 
     fn set_base_url(&self, base_url: Url) -> Result<()> {
-        *self.base_url.write()? = base_url;
+        let mut url = base_url;
+        if url
+            .query_pairs()
+            .filter(|(k, v)| k == "transport" && v == "polling")
+            .count()
+            == 0
```

Comment:
You could use `any` here which should be faster as it returns whenever it found an occurrence of the thing you're looking at, other that `filter`+ `count` which looks at every element.

Reply:
Good optimization!